packages/react-test-renderer/src/ReactTestRenderer.js

```diff
@@ -553,60 +549,57 @@ const ReactTestRendererFiber = {
   unstable_batchedUpdates: batchedUpdates,
   /* eslint-enable camelcase */
 
-  act(callback: () => void): Thenable {
+  act(callback: () => void | Promise<void>) {
     // note: keep these warning messages in sync with
-    // createNoop.js and ReactTestUtils.js
-    let result = batchedUpdates(callback);
-    if (__DEV__) {
-      if (result !== undefined) {
-        let addendum;
-        if (result !== null && typeof result.then === 'function') {
-          addendum =
-            "\n\nIt looks like you wrote TestRenderer.act(async () => ...) or returned a Promise from it's callback. " +
-            'Putting asynchronous logic inside TestRenderer.act(...) is not supported.\n';
-        } else {
-          addendum = ' You returned: ' + result;
+    // createReactNoop.js and ReactTestUtils.js
+    const result = actedUpdates(callback);
+    if (
+      result !== null &&
+      typeof result === 'object' &&
+      typeof result.then === 'function'
+    ) {
+      let called = false;
+      if (__DEV__) {
+        if (typeof global.Promise !== 'undefined') {
```

Comment:
This will crash outside of Node right? There's no `global` elsewhere. Not sure if people run test renderer in the browser but they might.

Reply:
If you're just trying to appease flow, add a suppression. These types aren't used anywhere anyway.